src/CMake/VisItMacros.cmake

```diff
@@ -263,3 +263,22 @@ function(ADD_CMAKE_GEN_TARGET gen_name
 endfunction()
 
 
+##############################################################################
+# This macro appends to a CACHE var list denoted by the NAME argument.
+# Caller is responsible for unsetting the CACHE var when no longer needed
+# to avoid polluting the CACHE.
```

Comment:
For the less CMake-inclined, what is the actually CMake syntax/function to do that?

Reply:
exactly what you see on this line:
https://github.com/visit-dav/visit/blob/374d3d567254702098526c440c6776f9d21cb555/src/CMake/VisItMacros.cmake#L282

I would have preferred to use `list(APPEND ...)`, and keep all lists in local scope to the PARENT dir, but the only way for a SUBDIR to set a variable that could be used by the PARENT dir and PEER dirs is with `set(... CACHE FORCE)`.
I created the macro to make things easier to read and understand.

Reply:
BTW, this was all necessitated by the fact that `blt_add_library` requires the SOURCES list.
With straight CMake, `add_library` can be called without any sources, then subdirs can add to a library's source list by use of `target_sources` command.

Reply:
Well, is that a strength or weakness of BLT? I mean, does BLT's appraoch somehow fix or prevent issues that CMake's allows/enables or vice versa?

Reply:
I think BLT makes some things with CMake simpler.  Other things not so much. Could be just the complexity of VisIt and use-cases they haven't encountered yet.  They do some bookkeeping and extra checks under the covers that could be problematic. I encountered an issue with OSPRAY_LIBRARIES yesterday I am trying to figure out the best way to work around. I could just not add it to the DEPENDS_ON list, but call target_link_libraries directly to avoid the issue BLT introduces.